daemon/cluster/executor/container/controller.go

```diff
@@ -369,20 +369,17 @@ func (r *controller) Shutdown(ctx context.Context) error {
 	}
 
 	if err := r.adapter.shutdown(ctx); err != nil {
-		if isUnknownContainer(err) || isStoppedContainer(err) {
-			return nil
+		if !(isUnknownContainer(err) || isStoppedContainer(err)) {
+			return err
 		}
-
-		return err
 	}
 
 	// Try removing networks referenced in this task in case this
 	// task is the last one referencing it
 	if err := r.adapter.removeNetworks(ctx); err != nil {
-		if isUnknownContainer(err) {
-			return nil
+		if !isUnknownContainer(err) {
```

Comment:
Why are we looking for unknown container here?

Reply:
this was already part of the original code, I'm not aware of the exact case for why it was needed

Reply:
looks like it was introduced by https://github.com/moby/moby/commit/534a90a99367af6f6bba1ddcc7eb07506e41f774